misc/local-dev.sh

```diff
@@ -0,0 +1,268 @@
+get_ifname() {
+    case "$(uname -s)" in
+        Darwin) echo lo0 ;;
+        Linux) echo lo ;;
+    esac
+}
+
+ifname_has_ip() {
+    local interface_name=$1
+    local ip=$2
+
+    case "$(uname -s)" in
+        Darwin)
+            ifconfig "${interface_name}" | grep -q "${ip}"
+            return $?
+            ;;
+
+        Linux)
+            ip addr show "${interface_name}" | grep -q "${ip}"
+            return $?
+            ;;
+    esac
+}
+
+ifname_add_ip() {
+    local interface_name=${1}
+    local ip=${2}
+
+    case "$(uname -s)" in
+        Darwin)
+            sudo ifconfig "${interface_name}" alias "${ip}/32"
+            ;;	
+
+        Linux)
+            sudo ip addr add "${ip}/32" dev "${interface_name}"
+            ;;
+    esac
+}
+
+ifname_del_ip() {
+    local interface_name=${1}
+    local ip=${2}
+
+    case "$(uname -s)" in
+        Darwin)
+            sudo ifconfig "${interface_name}" -alias "${ip}"
+            ;;
+
+        Linux)
+            sudo ip addr del "${ip}/32" dev "${interface_name}"
+            ;;
+    esac
+}
+
+render_config_template() {
+    local src=$1
+    local dst=$2
+    local fake_host_ip=$3
+    local host_port=${4:-"$TSURU_API_PORT"}
+
+    {
+        TSURU_HOST_IP="$fake_host_ip" TSURU_HOST_PORT="$host_port" \
+        envsubst < "$src" > "$dst"
+    }
+}
+
+
+# ------------------------------------------------------------------------------
+# Command execution functions
+#
+# These functions are the actual commands that are executed when the script is
+# run with a specific command.
+# ------------------------------------------------------------------------------
+exec_setup_loopback() {
+    local fake_host_ip=${1:-"$FAKE_HOST_IP"}
+    local ifname=$(get_ifname)
+
+    echo "Checking if the IP $fake_host_ip is assigned to the interface $ifname..."
+    ifname_has_ip "$ifname" "$fake_host_ip"
+    if [ $? -ne 0 ]; then
+        echo "Assigning the IP $fake_host_ip to the interface $ifname..."
+        ifname_add_ip "$ifname" "$fake_host_ip"
+    fi
+}
+
+exec_setup_tsuru_user() {
+    local user=${1:-"admin@admin.com"}
+    local password=${2:-"admin@123"}
+
+    echo "Setting up the root user with email $user and password $password..."
+
+    # Create the root user in Tsuru
+    # Ignore the output and errors because the user may already exist
+    echo -e "${password}\n${password}\n" | 
+        docker exec -i tsuru-api tsurud root user create "$user" &> /dev/null || true
+}
+
+exec_setup_tsuru_target() {
+    local target_host=${1:-"$FAKE_HOST_IP"}
+    local target_port=${2:-"$TSURU_API_PORT"}
+    local target_name=${3:-"local-dev"}
+
+    # Add the Tsuru target, ignore the output and errors because the target may already exist
+    echo "Setting up the Tsuru target $target_name at http://${target_host}:${target_port}..."
+    tsuru target add "$target_name" "http://${target_host}:${target_port}" &> /dev/null || true
```

Comment:
Something we talked about yesterday would be to add `--set-current` just as an extra safety measure

Reply:
Actually, breaking it in two commands would be better:
Leave the `tsuru target` as it is now
Add `tsuru target set "$target_name"` (so if the target already exists and the target adds fails, we still switch to development target after that)